### DISCUSSION TITLE
Implement navigation between matching keywords in the search results

### DISCUSSION THREAD

**Discussion by X1SDNA** [standardnotes/forum#2904] [Suggestions]
I have been a Standard Notes pro user for a few years (2 or 3 I believe).

I love it, and can't live without it. I recommend it to everyone. But there's an issue that I'd really appreciate a fix for ASAP.

I'm sure this has been brought up before, but it is a real problem that I deal with every day, to the point where unfortunately it pretty much defeats the purpose of having my notes encrypted.

I have many notes, some are long with lots of information. When I search for a term I need to find, SN will show me a list of notes in which that term exists. But that is where the search ends. I am not taken to the search terms in the notes. I need to manually scroll through all the information in the notes, and be on vigilant lookout for the yellow highlighted term. Because this is not realistically possible unless I'm sure what I'm looking for is at the very beginning of a note, I must select all, copy, open a text document in whatever desktop OS I am in, paste in in, run the search, be taken to the search term, and view/copy my information from that. Of course, I don't do this on the Android app, so I know I can't use the Android app if I need to search for something.

The obvious problem is that I am copying often sensitive information to the clipboard of my operating system all the time, which is why I mentioned that this largely defeats the purpose of having notes inaccessible to the OS.

Is there a better search function planned, one with which I can navigate to the location of each instance of the search terms found? And just in case some people are thinking it: Please don't tell me to make all my notes short, or other such workarounds.  I'd just really appreciate knowing why this doesn't work at all, or inconsistently (yes, I think remember an occasion or two where I have been taken to a search result in a note, but the last occurrence of that is not even in my memory, so maybe I simply can't recall.  I use SN on Linux, W10 and Android, so have been observing this for years now.

If searching can work like it does in Notepad, the Standard Notes will be pretty much perfect.

**Comment by effieeee** [comment]
Improvements to the search functionality have been surprisingly difficult to implement over the years, especially in light of the variety of note editors that can be loaded onto your notes 😞 We're keeping track of these requests on the forum though 🙏

**Comment by X1SDNA** [comment]
Hi - thanks for your reply. Can you implement it for one good markdown editor? I use Markdown Pro for everything, as it seems best suited to the type of note program that Standard Notes is.

It sounds like this situation is not likely to change anytime in the known future, and TBH that worries me. Needless to say, a good search function is fundamental to all data management and should be an absolute priority IMHO..

**Comment by myreli** [comment]
Hi @X1SDNA, thank you for your support for Standard Notes over the years. 

As @effieeee mentioned search has been historically hard to implement given the sandboxed nature of the editors. This is also the reason SN couldn't implement inline attachments and other features that integrated the editor with the app. 

To overcome this limitation the team has decided to invest time into building an in-house editor, called **Super**, which is integrated with the app like plaintext. So now we can choose Super, which is useful for text note-taking and offers this integrated experience, or the sandboxed editors, which are apps on themselves: Spreadsheets, Authenticator, and Kanban (community plugin).  Creating a good text editor is a whole challenge in itself, which is probably why it took a few years to embrace it. 

Super offers an out-of-the-box integrated experience and is now the flagship premium editor. It features search, Markdown Shortcuts, inline attachments, note-linking, and handling large files performantly, etc. Hopefully, this will fulfil your needs. Super is currently Beta, so a few things are still being worked on. For instance, you still need to search two times: first, outside of the editor to find the note; then inside the note. In the feature, we will certainly see improvements in the integration of Super with SN's ecosystem. 

I hope you enjoy experimenting with Super. But, most importantly, I hope it reassures you that SN is actively trying to improve the note-taking experience. 

<img width="844" alt="image" src="https://user-images.githubusercontent.com/17554234/221390881-a9f20221-1fbd-41ac-b2b6-e315baa6b491.png">
